aws/rust-runtime/aws-types/src/config.rs

```diff
@@ -60,6 +63,42 @@ impl Builder {
         self
     }
 
+    /// Set the retry_config for the builder
+    ///
+    /// # Examples
+    /// ```rust
+    /// use aws_types::config::Config;
+    /// use smithy_types::retry::RetryConfig;
+    /// let config = Config::builder().retry_config(RetryConfig::new()).build();
+    /// ```
+    pub fn retry_config(mut self, retry_config: impl Into<Option<RetryConfig>>) -> Self {
```

Comment:
This is interesting. In a lot of our other builders, we take a required arg for the non-prefixed setter, and then an option for the prefixed setter. I think I like this more though since it allows setting the option by reference or by owned value.

It doesn't look like this approach works well for string types though: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7dbc202c73345648d47ebdf562d368bf

Actually, just noticed the region one is like this too. @rcoh, do you have thoughts on this?